### DISCUSSION TITLE
Changing Zero Point

### DISCUSSION THREAD

**Discussion by 3DStyle** [arkypita/LaserGRBL#1108] [Q&A]
Is it possible to change the laser zero point?
As default LaserGRBL will start at the bottom left, like this:
![Point 1](https://user-images.githubusercontent.com/37050901/91272345-48205800-e784-11ea-99af-9fa215583f9e.png)

I want to have the control to change the zero point location, lets say I have a piece of wood that I want to laser a logo on its
center, so I'll be able to change the zero point in LaserGRBL to the center of the image, something that will look like this:

![2](https://user-images.githubusercontent.com/37050901/91273816-769f3280-e786-11ea-9fee-9a8ec35fc199.png)

**Comment by gmmanonymus111** [comment]
For images (png, jpg etc) you can do that when you are asked to set the size of your engraving. To set for example in the middle, the offsets should be negative values, and half the size of your engraving image. For a 50*50 cm image should have a -25 height and -25 widht offset.

For vectors, it depends on your editor program, but works on the same principle (offset should be half the size, in negative space)

**Comment by 3DStyle** [comment]
Thanks for the tip Alex, it works great!
However, an implemented button in LaserGRBL that will do this job for you will be more convenient.

**Comment by arkypita** [comment]
already in #64 as "Add image centring helper #90"
however i think you can create a custom button that set zero point at center of your job after import

**Comment by khturnings** [comment]
Hi
Sorry if this is the wrong place to add this request - 

I use SVG created in Inkscape for most of my laser work, and would find it very useful to set the zero point to centre of the SVG. This needs to take into account any boarder i.e. page size. As I will often, in Inkscape, make the page the size of the item to be engraved and then centre the logo I have drawn on the page.

Happy for this to be a button if this is easier.

Thanks

**Comment by Nordln** [comment]
When using Adobe Illustrator to export as SVG, the bottom-left of the artboard is position 0,0 when loaded into LaserGRBL. Simply position this corner at your home point. The fact that some of the image might be outside the artboard is ignored and the whole graphic shows in LaserGRBL.

**Comment by Wiki-the-Tinkerer** [comment]
I think it would be a very good idea to be able to select or define the workspaces' zero point. As an example I always have y homing point in the top right corner of the cnc, as that means the "tray is all out", so it's convinient to place the stock, while the laser/spindle is in the back corner.
